lib/index.ts

```diff
@@ -1,3 +1,22 @@
-import SchemaInspector from './schema-inspector';
+import Knex from 'knex';
+import {
+  SchemaInspector,
+  SchemaInspectorConstructor,
+} from './types/schema-inspector';
 
-export default SchemaInspector;
+export default function SchemaInspector(knex: Knex) {
+  let constructor: SchemaInspectorConstructor;
+
+  switch (knex.client.constructor.name) {
+    case 'Client_MySQL':
```

Comment:
we probably need to export dialect names as constants from knex

Reply:
Yeah I wasn't 100% sure if this was the way to do it accurately yet, any ideas on how to best retrieve this info from the Knex instance?